packages/console/src/components/Dropdown/use-position.ts

```diff
@@ -0,0 +1,43 @@
+import { RefObject, useCallback, useLayoutEffect, useState } from 'react';
+
+type Position = {
+  left: number;
+  top: number;
+  width: number;
+};
+
+const safePadding = 20;
+const overlayPadding = 4;
+
+export default function usePosition(
+  anchorRef: RefObject<HTMLElement>,
+  overlayRef: RefObject<HTMLElement>
+) {
+  const [position, setPosition] = useState<Position>();
+
+  const updatePosition = useCallback(() => {
+    if (anchorRef.current && overlayRef.current) {
+      const anchor = anchorRef.current.getBoundingClientRect();
+      const overlay = overlayRef.current.getBoundingClientRect();
+      const isTopSide =
+        anchor.y + anchor.height + overlay.height > window.innerHeight - safePadding;
+      const isLeftSide = anchor.x + overlay.width > window.innerWidth - safePadding;
```

Comment:
nit: `isOnTop` and `isOnLeft` would probably sound better

Reply:
updated